de.metas.shipper.gateway.dhl/src/main/java/de/metas/shipper/gateway/dhl/model/DhlClientConfigRepository.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * #%L
+ * de.metas.shipper.gateway.dhl
+ * %%
+ * Copyright (C) 2019 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.shipper.gateway.dhl.model;
+
+import de.metas.cache.CCache;
+import de.metas.util.Services;
+import org.adempiere.ad.dao.IQueryBL;
+import org.adempiere.exceptions.AdempiereException;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class DhlClientConfigRepository
+{
+	private final CCache<Integer, DhlClientConfig> cache = CCache.newCache(I_DHL_Shipper_Config.Table_Name, 1, CCache.EXPIREMINUTES_Never);
+
+	public DhlClientConfig getByShipperId(final int shipperId)
+	{
+		return cache.getOrLoad(shipperId, () -> readConfig(shipperId));
+	}
+
+	private static DhlClientConfig readConfig(final int shipperId)
```

Comment:
almost everywhere in the code we use the verb `retrieve` to describe a database read operation.
Pls use it to avoid having a spaghetti ubiquitous language.